Documentation for Subaccounts API (#1838)
Nexmo/nexmo-developer: _documentation/account/subaccounts/overview.md

---
title: Overview
meta_title: Subaccounts API (Beta)
navigation_weight: 1
description: The Subaccounts API (Beta) overview.
---

# Overview of Subaccounts API (Beta)

> **IMPORTANT:** Subaccounts API is released with **restricted availability**.

## Introduction

> The Subaccounts API allows to programmatically create and manage subaccounts for separate business units, use cases, product stages, or separate customers. The API empowers developers to handle various aspects of subaccount logistics: manage credit, track usage, set usage limits, suspend subaccounts, and so on.

Reviewer: > The Subaccounts API allows to programmatically create and manage subaccounts

Missing a word between "allows" and "to" here

![Subaccounts Overview](/assets/images/subaccounts/structure.png)

## Beta

This API is currently in Beta.

Nexmo always welcomes your feedback. Your suggestions help us improve the product. If you do need help, please email [support@nexmo.com](mailto:support@nexmo.com) and include the Subaccounts API in the subject line.

During Beta Nexmo will expand the capabilities of the API.

## Provisioning

Subaccounts API is released with **restricted availability**. To get access to the Subaccounts API please contact your account manager. [Nexmo Partners](https://info.nexmo.com/PartnerProgram.html) get access automatically.

## Supported features

In this release the following features are supported:

* Create a new subaccount
* Assign individual or shared balance to the subaccount
* Transfer credit (provided by Nexmo) from the primary account to the subaccount and vice versa
* Transfer balance from the primary account to the subaccount and vice versa
* List all subaccount balances and credits
* Show total credit and total balance across all subaccounts
* Suspend/re-activate the subaccount

## Concepts

### Account balance and credit

> All Nexmo accounts have an associated account balance. It shows the amount of funds available for spending. The account balance is deducted each time a chargeable API call is made. When account balance reaches zero, chargeable API calls cannot be executed unless the account has a credit line provided by Nexmo. If an account has a credit, then its account balance can go below zero. A postpaid account is an account that has credit, a prepaid account is an account without credit.

Reviewer: > A postpaid account is an account that has credit, a prepaid account is an account without credit.

Do you mean that a postpaid account has a credit _facility_ and prepaid accounts do not? If so, I think this needs to be mentioned here.

> RULE: Balance_available_for_transfer = |account_balance - credit_limit|

It means that the primary account can transfer funds to the secondary account, and these funds can come either from its balance (assuming it is positive) or from the credit provided by Nexmo. It is also possible to transfer balance from the subaccount back to the primary account, but direct transfer of balance between subaccounts is not supported.

Example: A postpaid primary account that initially had a zero balance and was given €100 in credit by Nexmo already used 20€ of the provided credit, that is, its balance was €-20. It had still €80 = |-20 - -100| that it could either spend itself or transfer to one of its subaccounts. The postpaid primary account decided to transfer €20 to Subaccount1. The primary account's balance became €-40 after this operation.

![Balance transfer](/assets/images/subaccounts/balance_transfer.png)

### Credit allocation

> A primary account is considered postpaid if it has a credit provided by Nexmo. The postpaid primary account can allocate a part of its credit to one of its subaccounts. Thus, it is possible to have a subaccount with a zero balance but non-zero credit. This subaccount will be able to make API calls until the allocated credit runs out. In general, any account that has a positive balance has `account_balance + |credit_limit|` funds available for spending.

Reviewer: Again, are we talking about a credit _facility_ here? To me, "credit" is just money in the account.

### Postpaid primary account

A postpaid primary account is responsible for spending of all of its subaccounts and its own spending (from the primary API key). The value that captures the total amount owned to Nexmo is the negative `total_balance` (positive `total_balance` means that nothing is owned to Nexmo). At the end of the month, Nexmo invoices a postpaid primary account for all usage on all its API keys (accounts) in that month. In theory, the total invoice amount would equal the `total_balance` across all accounts (all usage plus payments made).

Example: In the previous example, subaccount1's balance was 20, subaccount2's balance was 0, and the primary account's balance was -40. The total balance for the prepaid primary account and its subaccounts is: `total_balance = -40 + 20 + 0 = -20`.

## Best practices

* A Nexmo Partner should possess and manage a Nexmo primary account and should create subaccounts for its end customers.
* The Partner should not use its primary API key (account) to perform API calls. If the Partner wants to use Nexmo API itself, the Partner should create another subaccount.

> * When an end customer is close to using out all of its credit limit the Partner should either allocate additional credit limit to the end customer or wait for the end customer to pay before increasing end customer’s balance, otherwise the end customers API calls will be temporarily blocked.

Reviewer: ```suggestion
* When an end customer is close to reaching its credit limit the Partner ...
```
